plepy/helper.py

```diff
@@ -2,9 +2,13 @@
 
 
 def sigfig(n, sf: int):
-    mag = int(np.floor(np.log10(n)))
-    rf = sf - mag - 1
-    return round(n, rf)
+    if n == 0:
+        round_result = 0
+    else:
+        mag = int(np.floor(np.log10(abs(n))))
+        rf = sf - mag - 1
+        round_result = round(n,rf)
+    return round_result
```

Comment:
Can you tell me why this is needed?

Usually you keep full accuracy in the data when you are calculating it, and when you output it you just format it to the right number of significant digits with a format string, something like `"%4.2f" % my_num`.

Reply:
Monica can correct me if I explain this wrong, but the package calculates the confidence intervals for a a model that you are inputting into the package. For the package to run you provide the accuracy of the data you are inputting which is the significant figures. To calculate the confidence interval the criteria for the bounds uses convergence criteria  `ctol = sigfig(x_high, acc) - sigfig(x_low, acc)` so the sigfig function is used to ensure that the confidence intervals are calculated to the accuracy of the input data.

Reply:
We are performing a binary search, so it could run indefinitely (or at least out to Python's precision) unless we stop it. Since we could have different order of magnitude variables, we wanted a stopping criteria that was independent of magnitude.